### ISSUE TITLE
sound: create sandbox example for text to speech library

### ISSUE THREAD

**Issue by parkerbroz** [uchicago-cs/chiventure#1293]
assignees: Parker, Jason, Jake

summary: We aim to add a working example of our selected TTS library (espeak). We will do this by following the example demo from the espeak github repo by cloning the repo into chiventure and creating the test-espeak.c file. Hopefully, this will create text to speech sound coming from the VBox speakers on our laptops. We will also need to download the dependencies specified in the espeak github repo. In the future, we aim to generalize this process into functions so that other teams can use it and so that others can create games with text to speech.

**Comment by JaStarbuck** [comment]
sudo apt-get install espeak -y
I tried a few different ways to get espeak installed, this one was the first to work. Found on this website: https://www.howtoforge.com/how-to-install-espeak-on-ubuntu-20-04/

**Comment by JaStarbuck** [comment]
sudo snap install vlc
To do command line tts, a media player is needed. This command done in the command line downloads a mp4 player easily. From this website:
https://www.videolan.org/vlc/download-ubuntu.html

**Comment by JaStarbuck** [comment]
espeak "Hi this is a sample" -w sample.mp4 -g 60 -p 70 -s 100 -v en-us
Running this in the command line will output an .mp4 file and when opened with vlc, even in the vbox, sound is output. This is really cool! The next step is to get it to output sound from a text file or other type of file such as the one from the espeak github demo. Above command line is from this website:
https://www.howtoforge.com/how-to-install-espeak-on-ubuntu-20-04/

**Comment by parkerbroz** [comment]
I was able to mirror Jason's success using espeak on the command line. Finally getting my computer to talk was the most challenging and gratifying "hello world" I've ever done.
I was able to install and make espeak as a library, and even compiled a test file. However, when running the binary of the test file, I got errors eerily similar to the missing drivers one that haunted us when exploring SDL:
`shared memfd open() failed: Function not implemented`
`Saying  'Hello world!'...`
`shared memfd open() failed: Function not implemented`
`ALSA lib confmisc.c:767:(parse_card) cannot find card '0'`
`ALSA lib conf.c:4732:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory  `    
`ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings`
`ALSA lib conf.c:4732:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory`
`ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name`
`ALSA lib conf.c:4732:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory`
`ALSA lib conf.c:5220:(snd_config_expand) Evaluate error: No such file or directory`
`ALSA lib pcm.c:2642:(snd_pcm_open_noupdate) Unknown PCM default`
`error: No such file or directory`
`Done`

**Comment by JaStarbuck** [comment]
I got parker to go on my computer and do the same process so his driver problem wouldn't happen and it worked! When in the directory, all you have to do is run ./test-espeak and the machine through VBox will say "Hello World." So cool.

**Comment by JaStarbuck** [comment]
We also played around a little bit, you can change the text from within the test-espeak.c file, re do the *gcc test-espeak.c -lespeak-ng -o test-espeak* command, and then re run ./test-espeak and it will say the new text easy as that. I will now submit a pull request.

**Comment by JaStarbuck** [comment]
closing statment: Throughout this issue, we researched online some different ways to use the espeak library. First, we were able to get text from the command line to go through the speakers. Next, we followed the instructions on the espeak github to get text to speach from a .c file after cloning the espeak repo into the chiventure repo. We believe that it will be easy to take the example from the demo and integrate the process into other areas of chiventure. We will also have to get the dependencies for espeak downloaded into the linux machines.
